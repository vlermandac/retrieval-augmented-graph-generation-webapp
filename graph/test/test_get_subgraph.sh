#!/bin/bash

../KnowledgeGraph &
SERVER_PID=$!

sleep 2

JSON=$(cat <<EOF
{
  "index": "test_get_subgraph",
  "triplet_lists": [
    {
      "id": 1,
      "triplets": [
        {"entity1": "A", "relation": "B", "entity2": "C"},
        {"entity1": "C", "relation": "D", "entity2": "E"}
      ]
    },
    {
      "id": 2,
      "triplets": [
        {"entity1": "A", "relation": "B", "entity2": "E"}
      ]
    }
  ]
}
EOF
)

curl -X POST -H "Content-Type: application/json" -d "$JSON" http://localhost:8080/generate

# getting subgraph where the edges id is 2
curl -X POST -H "Content-Type: application/json" -d '{"index": "test_get_subgraph", "values": [2]}' http://localhost:8080/get > tmp_subgraph.json

sed '/"x":/d;/"y":/d;/"size"/d;' "tmp_subgraph.json" > "output.json"

expected_output=../../test/expected/test_get_subgraph.json
sed '/"x":/d;/"y":/d;/"size"/d;' "$expected_output" > "expected_output.json"

if diff -B -Z "expected_output.json" "output.json"; then
  echo "TestGet passed"
  kill $SERVER_PID
  rm -rf ../../data/test_get_subgraph
  exit 0
else
  echo "TestGet failed: files are different"
  kill $SERVER_PID
  rm -rf ../../data/test_get_subgraph
  exit 1
fi
