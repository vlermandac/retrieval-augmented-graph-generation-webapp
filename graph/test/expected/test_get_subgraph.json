{
  "attributes": {},
  "edges": [
    {
      "attributes": {
        "chunk_id": 2,
        "color": "#626880",
        "forceLabel": false,
        "label": "B",
      },
      "source": "A",
      "target": "E"
    }
  ],
  "nodes": [
    {
      "attributes": {
        "color": "#babbf1",
        "id": 1,
        "label": "A",
      },
      "key": "A"
    },
    {
      "attributes": {
        "color": "#babbf1",
        "id": 3,
        "label": "E",
      },
      "key": "E"
    }
  ],
  "options": {
    "multi": true
  }
}
